File: 26_remove_dups_from_sorted_array/remove_dups_from_sorted_array.js
// JS: https://leetcode.com/problems/remove-duplicates-from-sorted-array/submissions/1181006835/
// TS: https://leetcode.com/problems/remove-duplicates-from-sorted-array/submissions/1181005450/
function removeDuplicatesFromSortedArr(nums) {
    let hash = {};
    let count = 0;

    for(let i = 0; i < nums.length; i++) {
        if(hash[nums[i]] === undefined) {
            hash[nums[i]] = nums[i];
            nums[count] = nums[i];
            count++;
        }
    }

    return count;
}

console.log(removeDuplicatesFromSortedArr([1,1,2]))
console.log(removeDuplicatesFromSortedArr([0,0,1,1,1,2,2,3,3,4]))
console.log(removeDuplicatesFromSortedArr([-1,0,0,0,0,3,3]))

File: 26_remove_dups_from_sorted_array/remove_dups_from_sorted_array.py
from typing import List

def remove_dups_from_sorted_array(nums: List[int]) -> int:
    hash = dict()
    count = 0

    for n in nums:
        if hash.get(n) == None:
            hash[n] = n
            nums[count] = n
            count += 1
    
    return count

print(remove_dups_from_sorted_array([1,1,2]))
print(remove_dups_from_sorted_array([0,0,1,1,1,2,2,3,3,4]))
print(remove_dups_from_sorted_array([-1,0,0,0,0,3,3]))